frontend/express/public/javascripts/countly/countly.views.js

```diff
@@ -3782,183 +3782,403 @@ window.DownloadView = countlyView.extend({
+                    );
+                }else{
+                    $("#report-refresh-option").removeClass("selected");
+                    $("#report-onetime-option").addClass("selected");
+                    
+                }
+            }
+        }
+        $("#report-widget-drawer").addClass("open");
+    },
+    initDrawer: function(){
+        var self = this; 
+
+        $('#report-widge-close').off("click").on("click", function () {
+            $("#report-widget-drawer").removeClass("open");
+        });
+        $("#report-global-option").off("click").on("click", function(){
+            $("#report-global-option").addClass("selected");
+            $("#report-private-option").removeClass("selected");
+            self.reporInputValidator();
+        })
+        $("#report-private-option").off("click").on("click", function(){
+            $("#report-private-option").addClass("selected");
+            $("#report-global-option").removeClass("selected");
+            self.reporInputValidator();
+        })
+
+        $("#report-onetime-option").off("click").on("click", function(){
+            $("#report-onetime-option").addClass("selected");
+            $("#report-refresh-option").removeClass("selected");
+            $("#report-period-block").css("display", "none");
+            self.reporInputValidator();
+        })
+        $("#report-refresh-option").off("click").on("click", function(){
+            $("#report-refresh-option").addClass("selected");
+            $("#report-onetime-option").removeClass("selected");
+            $("#report-period-block").css("display", "block");
+            self.reporInputValidator();
+        })
+        
+        $("#single-period-dropdown").clySelectSetItems([
+            { value: 'today', name: jQuery.i18n.map["taskmanager.last-today"]},
+            { value: '7days', name: jQuery.i18n.map["taskmanager.last-7days"]},
+            { value: '30days', name: jQuery.i18n.map["taskmanager.last-30days"]}
+        ]);
+
+        $("#single-period-dropdown").clySelectSetSelection("", jQuery.i18n.map["drill.select_a_period"] );
+
+        $("#report-name-input").off("keyup").on("keyup", function(){
+            self.reporInputValidator();
+        })
+        $("#report-desc-input").off("keyup").on("keyup", function(){
+            self.reporInputValidator();
+        })
+        $("#single-period-dropdown").off("cly-select-change").on("cly-select-change", function (e, selected) {
+            self.reporInputValidator();
+        })
+
+        $("#create-report").off("click").on("click", function(){
+            var report_id =  $("#current_report_id").text();
+            var canSubmit = self.reporInputValidator();
+            if(!canSubmit){
+                return;
+            }
+            var report_name = $("#report-name-input").val();
+            var report_desc = $("#report-desc-input").val();
+            var global = $("#report-global-option").hasClass("selected");
```

Comment:
in theory global is not a reserved word, but very close to it, as could become in future, etc. 
Only for variable can we use something else? Like globalAccess.
It is ok for .global properties, no need to change there

Reply:
Updated